pkg/controller/context.go

```diff
@@ -9,8 +9,7 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-func (c *Controller) NewContext(req ctrl.Request) context.Context {
-	ctx := context.TODO()
+func (c *Controller) NewContext(ctx context.Context, req ctrl.Request) context.Context {
```

Comment:
IMO `NewContext` expecting a context as first parameter is odd.
I suggest to rename it.
```suggestion
func (c *Controller) PopulateContext(ctx context.Context, req ctrl.Request) context.Context {
```

Reply:
ok